pennylane/templates/subroutines/trotter.py

```diff
@@ -490,3 +492,339 @@ def compute_decomposition(*args, **kwargs):
+            &\vdots \\
+            S_{m}(t) &= S_{m-2}(p_{m}t)^{2} \cdot S_{m-2}((1-4p_{m})t) \cdot S_{m-2}(p_{m}t)^{2},
+        \end{align}
+
+    where the coefficient is :math:`p_{m} = 1 / (4 - \sqrt[m - 1]{4})`. The :math:`m`th order,
+    :math:`n`-step Suzuki-Trotter approximation is then defined as:
+
+    .. math:: e^{iHt} \approx \left [S_{m}(t / n)  \right ]^{n}.
+
+    For more details see `J. Math. Phys. 32, 400 (1991) <https://pubs.aip.org/aip/jmp/article-abstract/32/2/400/229229>`_.
+
+    Suppose we have direct access to the operators which represent the exponentiated terms of
+    a hamiltonian:
+
+    .. math:: \{ \hat{U}_{j} = e^{i t O_{j}} | for j \in [1, N] \}.
+
+    Given a quantum circuit which uses these :math:`\hat{U}_{j}` operators to represents the
+    first order expansion :math:`S_{1}(t)`; this class expands it to any higher order Suzuki-Trotter product.
+
+    .. warning::
+
+        :code:`TrotterizedQfunc` requires that the input function has a very specific function signature.
+        The first argument should be a time parameter which will be modified according to the Suzuki-Trotter
+        product formula. The wires required by the circuit should be either the last explicit argument or the
+        first keyword argument. :code:`qfunc((time, arg1, ..., arg_n, wires=[...], kwarg_1, ..., kwarg_n))`
+
+    .. warning::
+
+        :code:`TrotterizedQfunc` currently does not support pickling. Instead please decompose the operation
+        first before attempting to pickle the quantum circuit.
+
+    Args:
+        time (float): the time of evolution, namely the parameter :math:`t` in :math:`e^{iHt}`
+        *trainable_args (tuple): the trainable arguments of the first-order expansion function
+        qfunc (Callable): the first-order expansion given as a callable function which queues operations
+        wires (Iterable): the set of wires the operation will act upon (should be identical to qfunc wires)
+        n (int): an integer representing the number of Trotter steps to perform
+        order (int): an integer (:math:`m`) representing the order of the approximation (must be 1 or even)
+        reverse (bool): if true, reverse the order of the operations queued by :code:`qfunc`
+        name (str): an optional name for the instance
+        **non_trainable_kwargs (dict): non-trainable keyword arguments of the first-order expansion function
+
+    Raises:
+        ValueError: A qfunc must be provided to be trotterized.
+
+    **Example**
+
+    .. code-block:: python3
+
+        from pennylane.templates.subroutines.trotter import TrotterizedQfunc
+
+        def first_order_expansion(time, theta, phi, wires=[0, 1, 2], flip=False):
+            "This is the first order expansion (U_1)."
+            qml.RX(time*theta, wires[0])
+            qml.RY(time*phi, wires[1])
+            if flip:
+                qml.CNOT(wires=wires[:2])
```

Comment:
Is this a correct example of a first-order expansion? Might be good to mention the operation that is Terotterized this way. Not a blocker though.

Reply:
first-order expansion is just the name given to any such qfunc. Here I just applied gates at random. It's not trivial to determine what the base hamiltonian is that it is Trotterizing.